README.md
# Nao - Tell me a story [![GitHub license](https://img.shields.io/github/license/mashape/apistatus.svg)]() [![Version] (https://img.shields.io/badge/Version-2.0.0-green.svg)]()
> Nao app "Tell me a story".

I am trying to make Nao tell a story like a parent would do to its child. The main goal of the project is to make the voice of NAO feel like natural through the tone of his voice.

There already is a similar app in the AskNao ($$$) program but the voice is pretty linear and sometimes it's hard to understand what the robot says. I guess it could even be incomprehensible for someone with hearing problem or a foreigner.

More info on [Aldebaran Community](https://community.aldebaran.com/en/content/tell-me-story "Aldebaran Community")

## Install notes
### The Python way
First change the IP adress in the code to the one given by your robot.
Then run the the python script in /src/ by running the following commands in your terminal:

    chmod +x Le\ vieux\ grand-père.py
    ./Le\ vieux\ grand-père.py --ip XXX.XXX.XXX.XXX --port XXXX

### Using Choregraphe
Simply load the project located in /pkg/.

## Changelog

[Changelog is available here](CHANGELOG.md "Changelog")

## License
[MIT](LICENSE.md  "MIT")


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.
This project follows the [Semantic Versioning](http://semver.org/ "Semantic Versioning").

## 2.0.0
- Major updates to the [Choregraphe project](pkg/ "Choregraphe project"):
  - A new story "Le papillion haut dans le ciel" is here !
  - An interactive menu lets you choose the story you want to hear.
  - Changed the architecture of the project.
- Changed the [Python script](src/ "Python script") name (the same features as the Choregraphe project will be here soon.)

## 1.2.1
- Shebang !

## 1.2.0
- Changed the usage of the script.

## 1.1.2
- Fixed a problem concerning proxyPosture.goToPosture()

## 1.1.1
- The robot goes back to a regular position !

## 1.1.0
- Add gestures
- Slowed down the speech
- Minor speech improvements

## 1.0.0
- First version
  - tell the story "Le vieux grand-père"


src/Le vieux grand-père.py
# -*- coding: utf-8 -*-
from naoqi import ALProxy
import argparse

"""
\pau=xxx\    --> pause time (in ms)
\vct=xxx\    --> manage voice pitch 50 to 200% (default 100%)
\rspd=xxx\   --> voice speed 50 to 400% (default 100%)
\vol=xxx\    --> manage volume 1to 100%
"""
def main (IP, PORT=9559):
    # Declare the proxies we will be using
    atts = ALProxy("ALAnimatedSpeech", IP, PORT)
    motion = ALProxy("ALMotion", IP, PORT)
    proxyPosture = ALProxy("ALRobotPosture", IP, PORT)
    # Wake up the robot to make it able to move (enable the motors)
    motion.wakeUp()
    # Configure the body language
    configuration = {"bodyLanguageMode":"contextual"}
    # Text to be read
    atts.say("Bonjour, je vais te raconter l'histoire du vieux grand-père.\\pau=1500\\")
    atts.say("\\rspd=90\\Il était une fois un pauvre homme bien vieux, \\pau=200\\qui avait les yeux trouble, \\pau=200\\"\
            "l’oreille dure et les genoux tremblants. \\pau=600\\Quand il était à table,\\pau=200\\ il pouvait à peine tenir "\
            "sa cui yère ;\\pau=450\\ il reaipandait de la soupe sur la nappe,\\pau=200\\ et quelquefois mème en laissait échapper "\
            "de sa bouche.\\pau=600\\ La femme de son fils,\\pau=200\\ et son fils lui-mème en avaient pris un grand \\rspd=85\\daigoût, \\pau=200\\"\
            "\\rspd=100\\et à la fin ils le reléguaire dans un coin derrière le poil, \\pau=200\\ où ils lui donnaient à manger une "\
            "chaitive pitance dans une vieille écuelle de terre.\\pau=600\\ Le vieillard avaient souvent les larmes aux yeux,\\pau=200\\ "\
            "et regardait tristement du coter de la table...\\pau=800\\", configuration)
    atts.say("\\rspd=90\\Un jour, \\pau=200\\ l'écuelle,\\pau=200\\ que tenaient mal ses mains tremblantes,\\pau=200\\ tomba À terre et se brisa."\
            "\\pau=600\\La jeune femme s'emporta en reproches:\\pau=400\\ il n'osa rien répondre et baissa la tête en soupirant.\\pau=600\\"\
            "On lui \\rspd=90\\acheuta \\rspd=100\\ pour deux liards une écuelle de bois dans laquelle désormais on lui donnait à manger."\
            "\\pau=800\\", configuration)
    atts.say("\\rspd=90\\Quelques jours après,\\pau=200\\ son fils et sa belle-fille virent leur enfant qui avait quatre ans,\\pau=200\\ occupé \\pau=120\\"\
            "à assembler par terre de petites planchettes.\\pau=600\\ \\rspd=95\\ \\vct=70\\Que fais-tu là ?\\vct=100\\ \\rspd=100\\ Demanda "\
            "son père.\\pau=600\\ \\vct=150\\C'est un auget, \\pau=200\\ \\vct=100\\ répondit l'enfant,\\pau=200\\ \\vct=150\\pour donner À manger "\
            "à papa et \\rspd=85\\maman quand ils seront vieux.\\vct=100\\ \\pau=800\\", configuration)
    atts.say("\\rspd=90\\Le mari et la femme se regardère un instant sans rien dire,\\pau=200\\ puis ils se mirent à pleurer, \\pau=200\\ reprirent le "\
            "vieux grand-père à table,\\pau=200\\ et désormais le firent toujours manger à table avec eux,\\pau=200\\ sans plus jamais le rudoyer.", configuration)
    # Get the current posture of the robot
    posture = proxyPosture.getPostureFamily()
    # Set the robot to a 'Standard posture'
    if posture == "Standing":
            proxyPosture.goToPosture("Stand", 0.5)
    else:
            proxyPosture.goToPosture("Sit", 0.5)
    # Make it rest (disable the motors)
    motion.rest()

    print("done \n")

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--ip", type=str,
                        help="Robot ip address")
    parser.add_argument("--port", type=int,
                        help="Robot port number (Default: 9559)")

    args = parser.parse_args()
    main(args.ip, args.port)


src/Tell_me_a_story.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

from naoqi import ALProxy
import argparse

"""
\pau=xxx\    --> pause time (in ms)
\vct=xxx\    --> manage voice pitch 50 to 200% (default 100%)
\rspd=xxx\   --> voice speed 50 to 400% (default 100%)
\vol=xxx\    --> manage volume 1to 100%
"""
def main(IP, PORT=9559):
    
    # Declare the proxies we will be using
    atts = ALProxy("ALAnimatedSpeech", IP, PORT)
    motion = ALProxy("ALMotion", IP, PORT)
    proxyPosture = ALProxy("ALRobotPosture", IP, PORT)
    # Wake up the robot to make it able to move (enable the motors)
    motion.wakeUp()
    # Configure the body language
    configuration = {"bodyLanguageMode":"contextual"}
    # Text to be read
    atts.say("Bonjour, je vais te raconter l'histoire du vieux grand-père.\\pau=1500\\")
    atts.say("\\rspd=90\\Il était une fois un pauvre homme bien vieux, \\pau=200\\qui avait les yeux trouble, \\pau=200\\"\
            "l’oreille dure et les genoux tremblants. \\pau=600\\Quand il était à table,\\pau=200\\ il pouvait à peine tenir "\
            "sa cui yère ;\\pau=450\\ il reaipandait de la soupe sur la nappe,\\pau=200\\ et quelquefois mème en laissait échapper "\
            "de sa bouche.\\pau=600\\ La femme de son fils,\\pau=200\\ et son fils lui-mème en avaient pris un grand \\rspd=85\\daigoût, \\pau=200\\"\
            "\\rspd=100\\et à la fin ils le reléguaire dans un coin derrière le poil, \\pau=200\\ où ils lui donnaient à manger une "\
            "chaitive pitance dans une vieille écuelle de terre.\\pau=600\\ Le vieillard avaient souvent les larmes aux yeux,\\pau=200\\ "\
            "et regardait tristement du coter de la table...\\pau=800\\", configuration)
    atts.say("\\rspd=90\\Un jour, \\pau=200\\ l'écuelle,\\pau=200\\ que tenaient mal ses mains tremblantes,\\pau=200\\ tomba À terre et se brisa."\
            "\\pau=600\\La jeune femme s'emporta en reproches:\\pau=400\\ il n'osa rien répondre et baissa la tête en soupirant.\\pau=600\\"\
            "On lui \\rspd=90\\acheuta \\rspd=100\\ pour deux liards une écuelle de bois dans laquelle désormais on lui donnait à manger."\
            "\\pau=800\\", configuration)
    atts.say("\\rspd=90\\Quelques jours après,\\pau=200\\ son fils et sa belle-fille virent leur enfant qui avait quatre ans,\\pau=200\\ occupé \\pau=120\\"\
            "à assembler par terre de petites planchettes.\\pau=600\\ \\rspd=95\\ \\vct=70\\Que fais-tu là ?\\vct=100\\ \\rspd=100\\ Demanda "\
            "son père.\\pau=600\\ \\vct=150\\C'est un auget, \\pau=200\\ \\vct=100\\ répondit l'enfant,\\pau=200\\ \\vct=150\\pour donner À manger "\
            "à papa et \\rspd=85\\maman quand ils seront vieux.\\vct=100\\ \\pau=800\\", configuration)
    atts.say("\\rspd=90\\Le mari et la femme se regardère un instant sans rien dire,\\pau=200\\ puis ils se mirent à pleurer, \\pau=200\\ reprirent le "\
            "vieux grand-père à table,\\pau=200\\ et désormais le firent toujours manger à table avec eux,\\pau=200\\ sans plus jamais le rudoyer.", configuration)
    # Get the current posture of the robot
    posture = proxyPosture.getPostureFamily()
    # Set the robot to a 'Standard posture'
    if posture == "Standing":
            proxyPosture.goToPosture("Stand", 0.5)
    else:
            proxyPosture.goToPosture("Sit", 0.5)
    # Make it rest (disable the motors)
    motion.rest()

    print("done \n")

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--ip", type=str,
                        help="Robot ip address")
    parser.add_argument("--port", type=int,
                        help="Robot port number (Default: 9559)")

    args = parser.parse_args()
    main(args.ip, args.port)
